app/controllers/application_controller.rb

```diff
@@ -1,7 +1,7 @@
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
-  protect_from_forgery prepend: true, with: :exception unless ENV['DISABLE_CSRF'] == '1'
+  skip_forgery_protection if ENV['DISABLE_CSRF'] == '1'
```

Comment:
check if default behaviour is `with: :null_session` rather than  `with: :exception` and the impact of this

Reply:
appears to be defaulting to raising `ActionController::InvalidAuthenticityToken` which because we have a `rescue_from ActionController::InvalidAuthenticityToken` in application_controller it redirects back to referrer or unauthenticated root path

Reply:
as covered here https://github.com/rails/rails/pull/29742/files#diff-41177c56a83ce1447bfba614c4b7f29f3deef57f2312a9e0591763c489e76099R73-R79